### DISCUSSION TITLE
Comparison with alternatives

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/martin#1349] [Q&A]
How does Martin compare to alternatives? And what are the alternatives? For example, do you consider Martin, Tegola an TRex to be in the same category of vector tile servers?

**Comment by nyurik** [comment]
There are many alternatives, with somewhat overlapping goals and features.  We may want to create a book page comparing all other solutions. Feel free to start a page as a PR, and I will add more details to it.

See also (this list should grow):
* https://gis.stackexchange.com/questions/435951/what-is-the-difference-between-pg-tileserv-and-martin
* ...

**Comment by jjcfrancisco** [comment]
> How does Martin compare to alternatives? And what are the alternatives? For example, do you consider Martin, Tegola an TRex to be in the same category of vector tile servers?

One thing I can say about Martin vs others is that Martin is able to load data from a variety of sources. For instance, TRex is a PG only tile server. Mbtileserver is a mbtiles only tile server. And the list goes on. As far as I am aware, Martin is the only one that can serve tiles from a PG database, MBTiles or PMTiles file. This was a deal breaker at our company.

**Reply by nyurik** [reply]
Thanks! It took me a while to add that, e.g. in https://github.com/maplibre/martin/pull/456 ... Martin used to only support a single PG source only.
